dandischema/models.py

```diff
@@ -1116,19 +1116,38 @@ class BareAsset(CommonModel):
     }
 
     @validator("digest")
-    def digest_etag(cls, values):
-        try:
-            digest = values[DigestType.dandi_etag]
-            if "-" not in digest or len(digest.split("-")[0]) != 32:
-                raise ValueError
-        except KeyError:
-            raise ValueError("Digest is missing dandi-etag value.")
-        except ValueError:
-            raise ValueError(
-                f"Digest must have an appropriate dandi-etag value. "
-                f"Got {values[DigestType.dandi_etag]}"
-            )
-        return values
+    def digest_check(cls, values):
+        digest = values.get(DigestType.dandi_etag, None)
+        if digest is not None:
+            try:
+                if "-" not in digest or len(digest.split("-")[0]) != 32:
```

Comment:
You know there's [a regex for Dandi etags](https://github.com/dandi/dandischema/blob/0099e5f0c3a695db2db8336a13553c9fd0be9ff2/dandischema/digests/dandietag.py#L95), right?  Why don't you test against that?